### ISSUE TITLE
Question: Client channel handling with `mpsc::Sender<..>::try_send`

### ISSUE THREAD

**Issue by flxo** [bschwind/mqtt-broker#26]
Hello,

by reading the broker code I stumbled across all those `try_send` calls when the broker sends a packet to the client tasks. Almost everywhere the result is ignored which means that if the channel is full (e.g slow client or high load) the packet is just dropped. The channels to the clients are also configured with a buffer of 5 which immediately leads to packet loss when a client connects and sends with a higher in flight setting.

What is the desired approach here? 

I did a experiment and replaced the `try_send` with a `send(..).await` in order to perform some throughput tests. This backpressures between *everything* but is fine for my scenario (closed system with a limited number of well known clients).

**Comment by bschwind** [comment]
Backpressure is likely the desired outcome here. To be honest the `try_send()` calls are more laziness on my part as I read the spec and implement it properly. I really haven't done any work yet on high throughput capabilities or preventing packet loss. This is currently a hobby project so unfortunately I don't have a timeline for any of this, _but_ I'm very receptive to PRs!

**Comment by flxo** [comment]
Thanks for your reply! I pushed my hack (with some `expects` that are definitely wrong) [here](https://github.com/flxo/mqtt-broker/commit/588ca0f00f2216d41319ccd0196df0c908221d59). I had to refactor the `matching_subscribers` function because closures can be `async` which is needed for the `send(..).await`.

I'll try to draft something - probably with some TODOs. There probably need to be at least timeouts etc...
